crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1409,9 +1425,11 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                 // non-trivial marker expression that provoked the resolver to fork.
                 // In that case, we should ignore any dependency that cannot possibly
                 // satisfy the markers that provoked the fork.
-                if !possible_to_satisfy_markers(markers, requirement) {
-                    trace!("skipping {requirement} because of context resolver markers {markers}");
-                    return false;
+                if let Some(markers) = markers {
+                    if !possible_to_satisfy_markers(markers, requirement) {
+                        trace!("skipping {requirement} because of context resolver markers {markers}");
+                        return false;
+                    }
```

Comment:
Hmmm. So I very intentionally used `MarkerTree` instead of `Option<MarkerTree>`. For example, in this routine, I don't think there is any semantic difference between `None` and `Some(MarkerTree::And(vec![]))`. They are redundant with one another, which I find confusing, because it injects an extra state into the type system that you need to think about. Moreover, there _are_ perhaps cases where "a marker that is always true" and a "possibly absent marker" are semantically distinct, and IMO that's when we should use `Option<MarkerTree>`.

Reply:
Do we ever get a fork with `Some(MarkerTree::And(vec![]))`? It seems to me that we shouldn't fork in that case since we're still doing a universal resolution.

Reply:
I don't believe so, but that invariant isn't encoded into the type system. It's more like it's manifest in the concept of forking itself. It's not something you can easily employ local reasoning about when looking at just the types.

I think this is a fine change for now. I think this change is probably better than the status quo. :-)

Reply:
I encoded it in the type system using `ResolverMarkers`. Probably needs a fresh read tomorrow morning, but the api feels a good bit better this way.